classes/DirtSoundLibrary.sc

```diff
@@ -44,10 +44,16 @@ DirtSoundLibrary {
 			"\nreplacing '%' (%)\n".postf(name, buffers[name].size);
 			this.freeSoundFiles(name);
 		};
-		event = this.makeEventForBuffer(buffer);
+		event = this.makeEventForBuffer(buffer, metaData);
 		buffers[name] = buffers[name].add(buffer);
 		bufferEvents[name] = bufferEvents[name].add(event);
-		metaData !? { metaDataEvents[name] = metaDataEvents[name].add(metaData) };
+		metaData !? {
+			if(metaDataEvents[name].isNil) {
+				metaDataEvents[name] = Dictionary[];
+			};
+			index = buffers[name].size - 1;
+			metaDataEvents[name].put(index, metaData);
```

Comment:
What is wrong with 
```
metaData !? { metaDataEvents[name] = metaDataEvents[name].add(metaData) };
```
? It should be right …

Reply:
your code replaces the the last item, instead of appending. Also, `metaDataEvents[name]`  should be an array, I think.

Reply:
my intention with the dictionary was to keep the metadata collection sparse rather than filling an array with nils in the case that most samples lack smpl metadata. in fact this does not replace any item - the dictionary key `index` is calculated from `buffers[name]` (array) but the collection that is written to is `metaDataEvents[name]` (dictionary), which at this point has no item at `index`.

Reply:
Ah I see, it is a dictionary. Then it might be better to use the buffer itself as the key: `metaDataEvents[name].put(buffers[name], metaData);` (if there is not more than one metadata entry per buffer name).

Reply:
OK, still I don't think you need to calculate an index. You could just take the buffer object itself?

Reply:
personally i tend to prefer keys to be simple values that require minimal dependencies to construct, as i find it makes debugging easier. it also gives a nice symmetry between `buffers[name][i]`, `bufferEvents[name][i]` and `metaDataEvents[name][i]`. but i am open to changing it if you think the buffer object would be better. how is comparison between Buffers defined, object identity?

Reply:
sorry for my late reply. Yes, I agree that the symmetry is good. Of course there is one hidden asymmetry that the underlying object is an array for bufferEvents and a dictionary for metaDataEvents. But we'd have that anyway, so I think it is fine. I'll look thtough the code.